### PR TITLE
Is.object.improvement

### PR DESCRIPTION
**Pull request by ironmaniiith**
This change looks fine to me, please check if this is ok.

### CODE REVIEW

**Comment by ironmaniiith** [comment]
Oops, sorry for the first commit, I'll fix it

**Comment by jdalton** [comment]
Can you split the doc tweak into a separate PR (or rebase if it's noise).

**Comment by ironmaniiith** [comment]
I'll fix it.

File: is.js [hunk 1]
```diff
@@ -153,8 +153,7 @@
 
     // is a given value object?
     is.object = function(value) {
-        var type = typeof value;
-        return type === 'function' || type === 'object' && !!value;
+        return Object(value) === value;
```
Comment (jdalton) [inline is.js]: I originally avoided this due to some old old V8 JIT bug but sure let's do it.

**Pull request merged**